applications/h264/grpc_h264_endoscopy_tool_tracking/cpp/cloud/app_cloud_pipeline.hpp

```diff
@@ -0,0 +1,128 @@
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef GRPC_H264_ENDOSCOPY_TOOL_TRACKING_CPP_APP_CLOUD_PIPELINE_HPP
+#define GRPC_H264_ENDOSCOPY_TOOL_TRACKING_CPP_APP_CLOUD_PIPELINE_HPP
+
+#include <gxf/core/entity.hpp>
+#include <holoscan/holoscan.hpp>
+#include <holoscan/operators/format_converter/format_converter.hpp>
+#include <holoscan/operators/holoviz/holoviz.hpp>
+
+#include <grpc_server.hpp>
+#include <lstm_tensor_rt_inference.hpp>
+#include <tool_tracking_postprocessor.hpp>
+
+#include "../gxf_imports.hpp"
+
+namespace holohub::grpc_h264_endoscopy_tool_tracking {
+
+using namespace holoscan;
+using namespace holoscan::ops;
+
+using holoscan::entity::EntityResponse;
+
+/**
+ * @class AppCloudPipeline
+ * @brief A class that represents the application pipeline for H264 endoscopy tool tracking.
+ *
+ * This class inherits from HoloscanGrpcApplication and is responsible for composing the pipeline
+ * for processing video frames and performing tool tracking using a series of operators.
+ *
+ * @note the `HoloscanGrpcApplication` base class composes the `GrpcServerRequestOp` and
+ * `GrpcServerResponseOp` operators to handle incoming requests and outgoing responses.
+ * It also configures the queues for handling requests and responses.
+ */
+class AppCloudPipeline : public HoloscanGrpcApplication {
+ public:
+  AppCloudPipeline(std::queue<std::shared_ptr<nvidia::gxf::Entity>> incoming_request_queue,
+                   std::queue<std::shared_ptr<EntityResponse>> outgoing_response_queue)
+      : HoloscanGrpcApplication(incoming_request_queue, outgoing_response_queue) {}
+
+  void compose() override {
+    // Call base class compose to initialize the queues.
+    HoloscanGrpcApplication::compose();
+
+    // Create the Endoscopy Tool Tracking (ETT) Pipeline similar to the regular ETT application.
+    auto response_condition = make_condition<AsynchronousCondition>("response_condition");
+    auto video_decoder_context = make_resource<VideoDecoderContext>(
+        "decoder-context", Arg("async_scheduling_term") = response_condition);
+
+    auto request_condition = make_condition<AsynchronousCondition>("request_condition");
+    auto video_decoder_request =
+        make_operator<VideoDecoderRequestOp>("video_decoder_request",
+                                             from_config("video_decoder_request"),
+                                             Arg("async_scheduling_term") = request_condition,
+                                             Arg("videodecoder_context") = video_decoder_context);
+
+    auto video_decoder_response = make_operator<VideoDecoderResponseOp>(
```

Comment:
Could you update this code with the recent perfomance changes made with https://github.com/nvidia-holoscan/holohub/pull/550. Especially using BlockMemoryPool (or RMMAllocator available since 2.6, since the metadata already requires 2.6) and CudaStreamPool.

Reply:
I have replaced `UnboundedAllocator` with `RMMAllocator`.